### ISSUE TITLE
Units start before database is ready (stf local)

### ISSUE THREAD

**Issue by neofreko** [openstf/stf#117]
I consistently getting this when running `stf local`

```
FTL/reaper 7584 [reaper001] Unable to load initial state { [ReqlOpFailedError: Index `present` was not found on table `stf.devices` in:
r.table("devices").getAll(true, {index: "present"})
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^]
  name: 'ReqlOpFailedError',
  msg: 'Index `present` was not found on table `stf.devices`.',
  frames: [],
  message: 'Index `present` was not found on table `stf.devices` in:\nr.table("devices").getAll(true, {index: "present"})\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^' }
```

Manually adding the index to the table fixed the problem.

Apparently `migrate` cli already exit before the index was fully created.

**Comment by sorccu** [comment]
I am aware of this bug, but have never been able to reproduce it myself. Anyway it should be easily solvable for doing an `indexWait()` in `migrate`. Are you using an HDD or an SSD on that machine?

**Comment by neofreko** [comment]
It's SDD on OSX.

I tried this, but no luck yet (pardon my lack of nodejs-fu):
```
return rTable.indexCreate.apply(rTable, args).run(conn)
      .then(function() {
          //log.info('options for table %s index %s', table, index)
          if (!options) {
            log.info('Waiting for table %s index %s', table, index)
            return Promise.resolve(rTable.indexWait(args[0]).run(conn).then(function () {
              log.info('Index "%s"."%s" finally created', table, index)
            }))
          } else {
            log.info('Index "%s"."%s" created', table, index)
          }
      })
      .catch(alreadyExistsError, function() {
        log.info('Index "%s"."%s" already exists', table, index)
        return Promise.resolve()
      })
      .catch(noMasterAvailableError, function() {
        return Promise.delay(1000).then(function() {
          return createIndex(table, index, options)
        })
      })
  }
```

The way we write `table.js` might also requires reordering to make sure we are not creating index that depends on another index before it's dependencies got resolved.

Log:
```
INF/db 8452 [*] Connecting to 127.0.0.1:28015
INF/db:setup 8452 [*] Database "stf" created
INF/db:setup 8452 [*] Table "users" created
INF/db:setup 8452 [*] Index "users"."adbKeys" created
INF/db:setup 8452 [*] Table "logs" created
INF/db:setup 8452 [*] Table "vncauth" created
INF/db:setup 8452 [*] Waiting for table vncauth index response
INF/db:setup 8452 [*] Index "vncauth"."responsePerDevice" created
INF/db:setup 8452 [*] Table "devices" created
FTL/cli:migrate 8452 [*] Migration had an error: ReqlOpFailedError: Index `response` was not found on table `stf.vncauth` in:
```

**Comment by LiufxLucky365** [comment]
sorry, I have no any knowledge about nodejs, how can I fix this bug.

I always meet this bug today suddenly

**Comment by sorccu** [comment]
This should be fixed now, in 66fbadbd5d3ba39d4ae38d73cc8159ec521ddc9b. Commit message was cut short because the ticket number accidentally went on a new line, and because it began with a #, it became a comment.

**Comment by neofreko** [comment]
This is the log from today's master

```
INF/db 6257 [*] Connecting to 127.0.0.1:28015
INF/db:setup 6257 [*] Database "stf" created
INF/db:setup 6257 [*] Table "vncauth" created
INF/db:setup 6257 [*] Index "vncauth"."response" created
INF/db:setup 6257 [*] Waiting for index "vncauth"."response"
INF/db:setup 6257 [*] Index "vncauth"."responsePerDevice" created
INF/db:setup 6257 [*] Waiting for index "vncauth"."responsePerDevice"
INF/db:setup 6257 [*] Table "logs" created
FTL/cli:migrate 6257 [*] Migration had an error: ReqlOpFailedError: Index `response` was not found on table `stf.vncauth` in:
r.table("vncauth").indexWait("response")
```

Not sure this is more into rethinkdb API issue. Since indexWait said index was ready when it's not -- fail when we start accessing the index in question. indexStatus also return the same state. So the appropriate check should be an actual query. I tried this on my local but I messed up my promises and it never resolve :crying_cat_face:
